test/e2e/service.go

```diff
@@ -40,6 +40,10 @@ import (
 	"k8s.io/kubernetes/pkg/util/wait"
 )
 
+// Maximum time a kube-proxy daemon on a node is allowed to not
+// notice a Service update, such as type=NodePort.
+const kubeProxyLagTimeout = 5 * time.Minute
```

Comment:
Yeah it is. Compare the timestamps from kube-proxy logs in these when you have time, https://github.com/kubernetes/kubernetes/issues/18210#issuecomment-175794436. I think we should do some sort of snapshotting of iptables rules, check all these endpoints and services are in there. I haven't dug too deep into these optimizations.